plugins/woocommerce-admin/client/marketing/data-multichannel/reducer.ts

```diff
@@ -55,6 +60,37 @@ export const reducer: Reducer< State, Action > = (
 					error: action.payload,
 				},
 			};
+
+		case TYPES.RECEIVE_CAMPAIGNS_SUCCESS:
+			return {
+				...state,
+				campaigns: {
+					perPage: action.meta.perPage,
+					pages: {
+						...state.campaigns.pages,
+						[ action.meta.page ]: {
+							data: action.payload,
+						},
+					},
+					total: action.meta.total,
+				},
+			};
+
+		case TYPES.RECEIVE_CAMPAIGNS_ERROR:
+			return {
+				...state,
+				campaigns: {
+					perPage: action.meta.perPage,
+					pages: {
+						...state.campaigns.pages,
+						[ action.meta.page ]: {
+							error: action.payload,
+						},
+					},
+					total: action.meta.total,
+				},
+			};
```

Comment:
We can reduce code duplication tweaking the `[ action.meta.page ]` see https://github.com/woocommerce/woocommerce/pull/36735/files#r1119804861

Reply:
This is addressed in [`9feaf55` (#36735)](https://github.com/woocommerce/woocommerce/pull/36735/commits/9feaf55a6070e11ceb8633d38f22e1b51fa645c9), as mentioned in https://github.com/woocommerce/woocommerce/pull/36735#discussion_r1122208649.